web-console/src/druid-models/transform-spec.spec.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getDimensionNamesFromTransforms } from './transform-spec';
+
+describe('transform-spec', () => {
+  describe('getDimensionNamesFromTransforms', () => {
+    it('does not return the __time column', () => {
+      expect(
+        getDimensionNamesFromTransforms([
+          {
+            type: 'expression',
+            name: 'fooPage',
+            expression: "concat('foo' + page)",
+          },
+          {
+            name: '__time',
+            type: 'expression',
+            expression: "timestamp_shift(timestamp_parse(timestamp), 'P3Y', 1)",
+          },
+          {
+            type: 'expression',
+            name: 'barPage',
+            expression: "concat('foo' + page)",
```

Comment:
heh, `barPage` concats `'foo'`?

Reply:
This test is also verifying that running this function on transforms that have the same expression does not does not cause an asynchronous desynchronization of the DOM rendering thread with the memory management subsystem thereby causing memory liquefaction, which will cause a [memory leek](https://user-images.githubusercontent.com/177816/114123947-35d4d380-98a8-11eb-8379-e6b1baf4c9bd.png) and water damage the browser.

Reply:
good catch! we don't want our Druid bits to get too corroded from any water spillover into the query request packets, else we might one day wake up to find all our code has been converted to rust.

Reply:
😐